packages/publint/src/index.d.ts

```diff
@@ -121,6 +121,20 @@ export type Message =
     >
   | BaseMessage<'LOCAL_DEPENDENCY'>
   | BaseMessage<'BIN_FILE_NOT_EXECUTABLE'>
+  | BaseMessage<
+      'IMPORTS_GLOB_NO_DEPRECATED_SUBPATH_MAPPING',
+      {
+        expectPath: string[]
+        expectValue: string
+      }
+    >
+  | BaseMessage<'IMPORTS_FIELD_INVALID', { suggestValue: string }>
+  | BaseMessage<'IMPORTS_VALUE_INVALID', { suggestValue: string }>
+  | BaseMessage<'IMPORTS_GLOB_NO_MATCHED_FILES'>
+  | BaseMessage<'IMPORTS_MODULE_SHOULD_BE_ESM'>
+  | BaseMessage<'IMPORTS_TYPES_SHOULD_BE_FIRST'>
+  | BaseMessage<'IMPORTS_MODULE_SHOULD_PRECEDE_REQUIRE'>
+  | BaseMessage<'IMPORTS_DEFAULT_SHOULD_BE_LAST'>
```

Comment:
Other than `IMPORTS_FIELD_INVALID`, these rules are basically just copied over from the exports rules. 

Alternatively the existing rules could just be renamed to work for both (e.g. `EXPORTS_GLOB_NO_DEPRECATED_SUBPATH_MAPPING` becomes `EXPORTS_OR_IMPORTS_GLOB_NO_DEPRECATED_SUBPATH_MAPPING`).

Reply:
I think having separate `EXPORTS_` and `IMPORTS_` should be fine 👍 One thing with `"imports"` is that its consumer is the library itself, so I think most checks here would be validation-only.

In that case we could probably skip:
- IMPORTS_MODULE_SHOULD_BE_ESM
- IMPORTS_TYPES_SHOULD_BE_FIRST
- IMPORTS_MODULE_SHOULD_PRECEDE_REQUIRE

Since types and dual-publishing are usually not dealt inside `"imports"`, so it could be useful to skip the extra work and potentially extra maintenance. Unless maybe there's actual usecase for types and dual publishing there that I'm not aware of. Usually it's testing `browser`, `node` environments, etc

What do you think? Would this be tricky to implement?

Reply:
I've skipped those checks for now and I'll do some testing to see if they're necessary.

Reply:
@bluwy I've put together an example of why it may still be valuable to check the format of files in the `"imports"` field. In this example, the exports support dual publishing but the imports only support ESM. The exported code is importing code using the private imports mapping. If a consumer of this library is written in CommonJS, they will see a warning message.

Library: https://gist.github.com/kirkeaton/88f8033488be02a03c058653f1117433
Consumer: https://gist.github.com/kirkeaton/99f47ca581740be520812bd57ed52b27

Reply:
Interesting, yeah I think you're right. I guess I'm thinking more of the usecase where `"imports"` is used so that a single environment file can import a module that supports dual environments, e.g. browser/node. But still if the library publishes dual formats, then they'd have setup dual environments and dual formats still.

In that case, I think we can support `IMPORTS_MODULE_SHOULD_BE_ESM` and `IMPORTS_MODULE_SHOULD_PRECEDE_REQUIRE` then. The types I guess may still not be needed here?

If you'd like, we can hold off the release until these rules are supported.